module/VuFindSearch/tests/unit-tests/src/VuFindTest/Command/RetrieveBatchCommandTest.php

```diff
@@ -0,0 +1,133 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Search
+ * @author   Sudharma Kellampalli <skellamp@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org
+ */
+namespace VuFindTest\Command;
+
+use PHPUnit\Framework\TestCase;
+use VuFindSearch\Command\RetrieveBatchCommand;
+use VuFindSearch\ParamBag;
+
+/**
+ * Unit tests for RetrieveBatchCommand.
+ *
+ * @category VuFind
+ * @package  Search
+ * @author   Sudharma Kellampalli <skellamp@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org
+ */
+class RetrieveBatchCommandTest extends TestCase
+{
+    /**
+     * Test RetrieveBatch with RetrieveBatchInterface
+     *
+     * @return void
+     */
+    public function testExecute(): void
+    {
+        $params = new ParamBag(['foo' => 'bar']);
+        $backendId = 'bar';
+        $ids = ["id1", "id2"];
+        $backend = $this->getMockBuilder(\VuFindSearch\Backend\Solr\Backend::class)
+            ->disableOriginalConstructor()->getMock();
+        $command = new RetrieveBatchCommand($backendId, $ids, $params);
+        $backend->expects($this->once())->method('getIdentifier')
+            ->will($this->returnValue($backendId));
+        $backend->expects($this->once())->method('retrieveBatch')
+            ->with(
+                $this->equalTo($ids),
+                $this->equalTo($params)
+            )->will($this->returnValue('result'));
+        $this->assertEquals('result', $command->execute($backend)->getResult());
+    }
+
+    /**
+     * Test RetrieveBatch without RetrieveBatchInterface
+     *
+     * @return void
+     */
+    public function testExecuteNotInstance(): void
+    {
+        $params = new ParamBag(['foo' => 'bar']);
+        $backendId = 'bar';
+        $ids = ["id1", "id2"];
+        $command = new RetrieveBatchCommand($backendId, $ids, $params);
+        $backend = $this->getMockBuilder(\VuFindSearch\Backend\BackendInterface::class)
+            ->disableOriginalConstructor()->getMock();
+        $rci = $this->getMockBuilder(\VuFindSearch\Response\RecordCollectionInterface::class)
+            ->disableOriginalConstructor()->getMock();
+        $record = $this->getMockBuilder(\VuFindSearch\Response\RecordInterface::class)
+        ->disableOriginalConstructor()->getMock();
+        $backend->expects($this->exactly(2))->method('retrieve')
+            ->withConsecutive(
+                [ $this->equalTo('id1'),$this->equalTo($params)],
```

Comment:
Minor style issues: extra space after opening bracket, missing space after comma in this line and next.